### PR TITLE
:book: update hetzner-baremetal guide

### PR DESCRIPTION
**Pull request by Sayanta66**
**What this PR does / why we need it**:

Docs restructuring which includes the rewriting of the quickstart guide by means of addition and eliminations. Baremetal and node image contents are separated. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits
- [x] include documentation
- [ ] add unit tests

### CODE REVIEW

File: docs/topics/quickstart.md [hunk 1]
```diff
@@ -1,10 +1,26 @@
 # Quickstart Guide
 
-This guide goes through all the necessary steps to create a cluster on Hetzner infrastructure (on HCloud & Hetzner Dedicated).
+This guide goes through all the necessary steps to create a cluster on Hetzner infrastructure (on HCloud).
 
-## Preparing Hetzner
+>Note: The cluster templates used in the repository and in this guide for creating clusters are for development purposes only. These templates are not advised to be used in the production environment. However, the software is production-ready and users use it in their production environment. Make your clusters production-ready with the help of Syself Autopilot. For more information, contact <info@syself.com>.
```
Comment (guettli) [inline docs/topics/quickstart.md:5]: I think a hyperlink to the Autopilot landing page would be good here. But afaik this URL does not exist yet.

Comment (janiskemper) [reply docs/topics/quickstart.md:5]: no it does not

Comment (kranurag7) [reply docs/topics/quickstart.md:5]: will update given the page is available now.

File: docs/topics/hetzner-baremetal.md [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+### Creating new user in Robot 
+To create new user in Robot, click on the `Create User` button in the Hetzner Robot console. Once you create the new user, a user ID will be provided to you via email from Hetzner Robot. The password will be the same that you used while creating the user.
+
+![robot user](../pics/robot_user.png)
+
+This is a required for following the next step.
+
+### creating and verify ssh-key in hcloud 
+```bash
+hcloud ssh-key create --name test --public-key-from-file ~/.ssh/test.pub
```
Comment (janiskemper) [inline docs/topics/hetzner-baremetal.md:119]: what is "test"? I think you forgot a step where you generate an ssh key called "test"

File: docs/topics/hetzner-baremetal.md [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+This is an important step as the ssh key is used to access the servers. Make sure you are using the correct ssh key name.
+
+The `test` is the name of the ssh key that we have created above. It is because the generated manifest references `test` as the ssh key name. 
+```yaml
+  sshKeys:
+    hcloud:
+    - name: test
+```
+
+### Create a secret for Hetzner (Hcloud + Robot)
```
Comment (janiskemper) [inline docs/topics/hetzner-baremetal.md:132]: We create two secrets here, not only the "hetzner" one

File: docs/topics/hetzner-baremetal.md [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+> NOTE: sshkey-name should must match the name that is present in hetzner otherwise the controller will not know how to reach the machine.
+
+Patch the created secrets so that they get automatically moved to the target cluster later. The following command helps you do that:
+
+```shell
+kubectl patch secret hetzner -p '{"metadata":{"labels":{"clusterctl.cluster.x-k8s.io/move":""}}}'
+kubectl patch secret robot-ssh -p '{"metadata":{"labels":{"clusterctl.cluster.x-k8s.io/move":""}}}'
+```
+
+The secret name and the tokens can also be customized in the cluster template.
```
Comment (janiskemper) [inline docs/topics/hetzner-baremetal.md:165]: doesn't the cluster object have some pre-conditions on how the ssh keys are named? Currently, this is not described. Probably we have hard-coded it on "test" right?

File: docs/topics/hetzner-baremetal.md [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+kubeadmcontrolplane.controlplane.cluster.x-k8s.io/my-cluster-control-plane created
+hcloudmachinetemplate.infrastructure.cluster.x-k8s.io/my-cluster-control-plane created
+hcloudmachinetemplate.infrastructure.cluster.x-k8s.io/my-cluster-md-0 created
+hcloudremediationtemplate.infrastructure.cluster.x-k8s.io/control-plane-remediation-request created
+hcloudremediationtemplate.infrastructure.cluster.x-k8s.io/worker-remediation-request created
+hetznerbaremetalmachinetemplate.infrastructure.cluster.x-k8s.io/my-cluster-md-1 created
+hetznercluster.infrastructure.cluster.x-k8s.io/my-cluster created
+```
+
+### Creating Object Manifests
```
Comment (janiskemper) [inline docs/topics/hetzner-baremetal.md:193]: It doesn't make sense IMO to do this after creating the cluster.. I think we should have the following order:
1. Preparing provider
2. Creating management cluster
3. Creating secrets in management cluster
4. Creating host objects in mgt cluster
5. Creating workload cluster

File: docs/topics/hetzner-baremetal.md [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: HetznerBareMetalHost
+metadata:
+  name: "caph-baremetal-server"
+  namespace: default
+spec:
+  description: CAPH BareMetal Server
+  serverID: <ID-of-your-server> # please check robot console
+  rootDeviceHints:
+    wwn: eui.002538bb31006f91
```
Comment (janiskemper) [inline docs/topics/hetzner-baremetal.md:207]: don't hard-code this. Make it at least with <>

**Review by janiskemper — CHANGES_REQUESTED** [review]

File: docs/topics/hetzner-baremetal.md (hunk 4 above)
Comment (kranurag7) [reply docs/topics/hetzner-baremetal.md:165]: We've hardcoded it in the template.

Comment (janiskemper) [reply docs/topics/hetzner-baremetal.md:165]: so mention it in the docs ;)

File: docs/topics/hetzner-baremetal.md [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+  name: "caph-baremetal-server"
+  namespace: default
+spec:
+  description: CAPH BareMetal Server
+  serverID: <ID-of-your-server> # please check robot console
+  maintenanceMode: false
+```
+In the above server, we have not specified the WWN of the server and we have applied it in the cluster. One HetznerBareMetalMachine will pick up this host and will start provisioning.
+
+```bash
```
[the remark this answers is not in the document]
Comment (kranurag7) [reply docs/topics/hetzner-baremetal.md:228]: this sentence or the scenario we consider where the wwn is not specified.

[the remark this answers is not in the document]
Comment (janiskemper) [reply docs/topics/hetzner-baremetal.md:228]: this is only about hetznerbaremetalmachine.. That's a side-note that I don't think is so important that we have to mention it in the quickstart, that's all!

File: docs/topics/hetzner-baremetal.md [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+$ kubectl get hetznerbaremetalhost -A
+default     my-cluster-md-1-tgvl5   my-cluster   default/test-bm-gpu    my-cluster-md-1-t9znj-694hs   Provisioning   23m   ValidationFailed   no root device hints specified
+```
+
+After you see the error, describe the `HetznerBareMetalHost` object and then you will find the `wwn` of the server in the status of the `HetznerBareMetalHost` resource.
+
+```yaml
+      Storage:
+        Hctl:             2:0:0:0
```
[the remark this answers is not in the document]
Comment (janiskemper) [reply docs/topics/hetzner-baremetal.md:242]: and it is not well readable

File: docs/topics/hetzner-baremetal.md [hunk 9]
```diff
@@ -305,6 +278,33 @@ sdd  disk <wwn>                  1.8T
 
 In the above case, you can use any of the four disks available to you on a baremetal server. 
 
+
+### Creating Workload Cluster 
+
+Since we have already created secret in hetzner robot, hcloud and ssh-keys as secret in management cluster, we can now apply the cluster.
```
Comment (janiskemper) [inline docs/topics/hetzner-baremetal.md:284]: can you mention again that secrets and ssh key names are hard-coded and have to be named properly? You mention it now shortly above but there is no connection that it is hard-coded here in the template

**Review by janiskemper — CHANGES_REQUESTED** [review]

File: docs/topics/hetzner-baremetal.md [hunk 10]
```diff
@@ -143,7 +141,8 @@ The `test` is the name of the ssh key that we have created above. It is because
 
 In order for the provider integration hetzner to communicate with the Hetzner API ([HCloud API](https://docs.hetzner.cloud/) + [Robot API](https://robot.your-server.de/doc/webservice/en.html#preface)), we need to create secrets with the access data. The secret must be in the same namespace as the other CRs.
 
-We create two secrets named `hetzner` for hcloud and `robot-ssh` for hetzner robot services.
+We create two secrets named `hetzner` for hcloud and `robot-ssh` for hetzner and robot services.
```
Comment (janiskemper) [inline docs/topics/hetzner-baremetal.md:144]:
```suggestion
We create two secrets named `hetzner` for Hetzner Cloud and Robot API access and `robot-ssh` for provisioning bare metal servers via SSH.
```

**Comment by janiskemper** [comment]
looks good!

**Review by kranurag7 — APPROVED** [review]

**Review by janiskemper — APPROVED** [review]

**Pull request merged**